subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/UploadToQapps.kt

```diff
@@ -34,3 +43,25 @@ class UploadToQapps(context: String, artifactsConfiguration: ArtifactsConfigurat
         }
     }
 }
+
+internal class UploadToQappsCondition(
+    val config: CdBuildConfig?
+) {
+
+    fun isReleaseChain(): Boolean {
+        return findDeployment()?.isRelease ?: false
+    }
+
+    fun canUpload(): Boolean {
+        if (config == null) return true
```

Comment:
Are you sure that this method should return `true` when `config` is absent?

Reply:
Yes, it means that this build is not a part of a CD release.
In this case, uploading to QApss works as described in build script.